app/controllers/articles_controller.rb

```diff
@@ -213,25 +225,39 @@ def article_params
     params.require(:article).permit(modified_params)
   end
 
+  # TODO: refactor all of this update logic into the Articles::Updater possibly,
```

Comment:
Yup, exactly why we have objects like this. No clean reason for the logic to live in the controllers.